kotlin-native/backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/objcexport/ObjCExportMapper.kt

```diff
@@ -97,9 +97,18 @@ internal fun ObjCExportMapper.shouldBeExposed(descriptor: CallableMemberDescript
     // KT-42641. Don't expose componentN methods of data classes
     // because they are useless in Objective-C/Swift.
     isComponentNMethod(descriptor) && descriptor.overriddenDescriptors.isEmpty() -> false
+    descriptor.isHiddenFromObjC() -> false
     else -> true
 }
 
+private fun CallableMemberDescriptor.isHiddenFromObjC(): Boolean = when {
+    // Note: the front-end checker requires all overridden descriptors to be either refined or not refined.
+    overriddenDescriptors.isNotEmpty() -> overriddenDescriptors.first().isHiddenFromObjC()
+    else -> annotations.any { annotation ->
+        annotation.annotationClass?.annotations?.any { it.fqName == KonanFqNames.hidesFromObjC } == true
```

Comment:
@SvyatoslavScherbina I am seeing some weird/unexpected behaviour. The built-in annotations are working great. However a custom annotation (either from a library, or in the same source) isn't recognised in the backend. The weird thing is that the annotations are correctly recognised by the frontend checkers (e.g. using an ObjC and Swift annotation results in an error). Any thoughts on possible causes? I can create a reproducer if needed.

Reply:
No, no thoughts, please investigate.

Reply:
My bad the custom annotation had the `Retention` set to `SOURCE` instead of `BINARY` 🤦🏻.
Everything works as expected!

Reply:
Ah, ok, makes sense :)
Thank you for the update!